homeassistant/components/conversation/default_agent.py

```diff
@@ -484,84 +495,93 @@ def _get_or_load_intents(
 
         if lang_intents is None:
             intents_dict: dict[str, Any] = {}
-            loaded_components: set[str] = set()
+            language_variant: str | None = None
         else:
             intents_dict = lang_intents.intents_dict
-            loaded_components = lang_intents.loaded_components
+            language_variant = lang_intents.language_variant
 
-        # en-US, en_US, en, ...
-        language_variations = list(_get_language_variations(language))
+        domains_langs = get_domains_and_languages()
 
-        # Check if any new components have been loaded
-        intents_changed = False
-        for component in hass_components:
-            if component in loaded_components:
-                continue
+        if not language_variant:
+            # Choose a language variant upfront and commit to it for custom
+            # sentences, etc.
+            all_language_variants = {
+                lang.lower(): lang for lang in itertools.chain(*domains_langs.values())
+            }
 
-            # Don't check component again
-            loaded_components.add(component)
+            # en-US, en_US, en, ...
+            for maybe_variant in _get_language_variations(language):
+                matching_variant = all_language_variants.get(maybe_variant.lower())
+                if matching_variant:
+                    language_variant = matching_variant
+                    break
 
-            # Check for intents for this component with the target language.
-            # Try en-US, en, etc.
-            for language_variation in language_variations:
-                component_intents = get_intents(
-                    component, language_variation, json_load=json_load
+            if not language_variant:
+                _LOGGER.warning(
+                    "Unable to find supported language variant for %s", language
                 )
-                if component_intents:
-                    # Merge sentences into existing dictionary
-                    merge_dict(intents_dict, component_intents)
+                return None
 
-                    # Will need to recreate graph
-                    intents_changed = True
-                    _LOGGER.debug(
-                        "Loaded intents component=%s, language=%s (%s)",
-                        component,
-                        language,
-                        language_variation,
-                    )
-                    break
+            # Load intents for all domains supported by this language variant
+            for domain in domains_langs:
+                domain_intents = get_intents(
```

Comment:
If we're going to load all languages at once, let's change the intent package to only ship a single JSON file per language

Reply:
OK, I've made a note of this for a future PR